### ISSUE TITLE
Placing Moloch Shell Fragment into toolbelt deletes it and spits out a bunch of errors

### ISSUE THREAD

**Issue by KingkumaArt** [Regalis11/Barotrauma#11765]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Not much more to explain, was looting a moloch, went to place a shell fragment in my toolbelt, and it just gave me about 20 red errors. It had all 4 drops, unsure if thats important.

### Reproduction steps

Kill moloch until shell dropped
Equip Toolbelt
Drag Shell onto toolbelt slot

### Bug prevalence

Just once

### Version

v1.0.8.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Something went wrong when trying to swap items between inventory slots: couldn't find the source item from it's inventory.
   at System.Environment.get_StackTrace()
   at Barotrauma.Inventory.TrySwapping(Int32 index, Item item, Character user, Boolean createNetworkEvent, Boolean swapWholeStack) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 696
   at Barotrauma.Inventory.TryPutItem(Item item, Int32 i, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 542
   at Barotrauma.CharacterInventory.TryPutItem(Item item, Int32 index, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\CharacterInventory.cs:line 452
   at Barotrauma.Inventory.UpdateDragging() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Inventory.cs:line 1263
   at Barotrauma.CharacterHUD.Update(Single deltaTime, Character character, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\CharacterHUD.cs:line 287
   at Barotrauma.Character.UpdateControlled(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\Character.cs:line 456
   at Barotrauma.Character.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 3178
   at Barotrauma.AICharacter.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AICharacter.cs:line 38
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 3037
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 186
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 881
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 93
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
```

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/0022d4b32d18c346380aede09dd807b62410f745

**Comment by Rokvach** [comment]
Seems to now cause a crash

```
Barotrauma Client crash report (generated on 05/08/2023 10:55:00)

Barotrauma seems to have crashed. Sorry for the inconvenience! 

B5AB0B16BAC449F5E8680446E0AD84DC

Game version 1.0.13.0 (ReleaseWindows, branch fix/misc-fixes-apr09, revision 58542148f0)
Graphics mode: 1280x720 (Windowed)
VSync OFF
Language: English
Selected content packages: Vanilla (FA145D2)
Level seed: no level loaded
Loaded submarine: Remora (1591D6A4E4E3D8B57962A2E639DF3EA4)
Selected screen: Barotrauma.GameScreen
SteamManager initialized

System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1.7777778}
    GPU status: Normal

Exception: Tried to add an item to a full inventory slot (stack already full). (System.InvalidOperationException)
Target site: Void Add(Barotrauma.Item)
Stack trace: 
   at Barotrauma.Inventory.ItemSlot.Add(Item item) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 140
   at Barotrauma.Inventory.ForceToSlot(Item item, Int32 index) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 956
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Barotrauma.Inventory.TrySwapping(Int32 index, Item item, Character user, Boolean createNetworkEvent, Boolean swapWholeStack) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 839
   at Barotrauma.Inventory.TryPutItem(Item item, Int32 i, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Inventory.cs:line 542
   at Barotrauma.CharacterInventory.TryPutItem(Item item, Int32 index, Boolean allowSwapping, Boolean allowCombine, Character user, Boolean createNetworkEvent, Boolean ignoreCondition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\CharacterInventory.cs:line 454
   at Barotrauma.Inventory.UpdateDragging() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Inventory.cs:line 1263
   at Barotrauma.CharacterHUD.Update(Single deltaTime, Character character, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\CharacterHUD.cs:line 293
   at Barotrauma.Character.UpdateControlled(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Characters\Character.cs:line 456
   at Barotrauma.Character.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 3178
   at Barotrauma.AICharacter.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AICharacter.cs:line 38
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 3037
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 186
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 881
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 93
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58

Last debug messages:
[05/08/2023 10:54:56] spawnitem "Moloch Shell Fragment" inventory
```

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
